docs/core/deploying/single-file.md

```diff
@@ -11,9 +11,26 @@ Bundling all application-dependent files into a single binary provides an applic
+## Impact of using SingleFile feature
 
-Single-file doesn't bundle native libraries by default. On Linux, we prelink the runtime into the bundle and only application native libraries are deployed to the same directory as the single-file app. On Windows, we prelink only the hosting code and both the runtime and application native libraries are deployed to the same directory as the single-file app. This is to ensure a good debugging experience, which requires native files to be excluded from the single file. There is an option to set a flag, `IncludeNativeLibrariesForSelfExtract`, to include native libraries in the single file bundle, but these files will be extracted to a temporary directory in the client machine when the single file application is run.
+There are some caveats that you need to be aware for single-file use, first of which is the use of path information to locate a file relative to the location of your application. The <xref:System.Reflection.Assembly.Location?displayProperty=nameWithType> API will return an empty string, which is the default behavior for assemblies loaded from memory. The compiler will give a warning for this API during build time to alert the developer to the specific behavior. If the path to the application directory is needed, the <xref:System.AppContext.BaseDirectory?displayProperty=nameWithType> API will return the directory where the AppHost (the single-file bundle itself) resides. The table at the bottom of this section has runtime library API behavior in single-file use.
+
+Single-file doesn't bundle native libraries by default. On Linux, we prelink the runtime into the bundle and only application native libraries are deployed to the same directory as the single-file app. On Windows, we prelink only the hosting code and both the runtime and application native libraries are deployed to the same directory as the single-file app. This is to ensure a good debugging experience, which requires native files to be excluded from the single file. There is an option to set a flag, `IncludeNativeLibrariesForSelfExtract`, to include native libraries in the single file bundle, but these files will be extracted to a temporary directory in the client machine when the single file application is run. Similarly, single-file doesn't bundle debug files (PDB files) for a better debugging experience.
```

Comment:
> Similarly, single-file doesn't bundle debug files (PDB files) for a better debugging experience.

This makes it sound like that not bunding PDB files improves debugging experience that's not correct.

Do we want to mentioned embedded PDBs?

Reply:
Yes, I would instead say something like, "to include PDBs inside the application, use the \<DebugType>embedded</> flag" +link

Reply:
We had a publish property for this in pre 5.0, IncludeSymbolsInSingleFile, but stopped supporting it in 5.0 with [this](https://github.com/dotnet/sdk/pull/12724/files#diff-7dc833d02f198e5a9f0ab88e229db18aR97) checkin.

@sbomer might have more context on the reason

Reply:
Looks like the DebugType route was the direction we wanted to go based on the github discussion on this